airflow/providers/yandex/secrets/secrets_manager.py

```diff
@@ -0,0 +1,280 @@
+    the path ``airflow/variables/hello``, the variable with the name ``hello`` would be accessible.
+
+    When ``{"config_prefix": "airflow/config"}`` is set, if a secret is defined with
+    the path ``airflow/config/sql_alchemy_conn``, the config with key ``sql_alchemy_conn`` would be
+    accessible.
+
+    When the prefix is empty, keys will use the Lockbox Secrets without any prefix.
+
+    .. code-block:: ini
+
+        [secrets]
+        backend = airflow.providers.yandex.secrets.secrets_manager.SecretsManagerBackend
+        backend_kwargs = {"yc_connection_id": "<connection_ID>", "folder_id": "<folder_ID>"}
+
+    You need to specify credentials or id of yandexcloud connection to connect to Yandex Lockbox with.
+    Credentials will be used with this priority:
+
+    * OAuth Token
+    * Service Account JSON file
+    * Service Account JSON
+    * Yandex Cloud Connection
+
+    If no credentials specified, default connection id will be used.
+
+    Also, you need to specify the Yandex Cloud folder ID to search for Yandex Lockbox secrets in.
+
+    :param yc_oauth_token: Specifies the user account OAuth token to connect to Yandex Lockbox with.
+        Looks like ``y3_xxxxx``.
+    :param yc_sa_key_json: Specifies the service account auth JSON.
+        Looks like ``{"id": "...", "service_account_id": "...", "private_key": "..."}``.
+    :param yc_sa_key_json_path: Specifies the service account auth JSON file path.
+        Looks like ``/home/airflow/authorized_key.json``.
+        File content looks like ``{"id": "...", "service_account_id": "...", "private_key": "..."}``.
+    :param yc_connection_id: Specifies the connection ID to connect to Yandex Lockbox with.
+        Default: "yandexcloud_default"
+    :param folder_id: Specifies the folder ID to search for Yandex Lockbox secrets in.
+        If set to None (null in JSON), requests will use the connection folder_id if specified.
+    :param connections_prefix: Specifies the prefix of the secret to read to get Connections.
+        If set to None (null in JSON), requests for connections will not be sent to Yandex Lockbox.
+        Default: "airflow/connections"
+    :param variables_prefix: Specifies the prefix of the secret to read to get Variables.
+        If set to None (null in JSON), requests for variables will not be sent to Yandex Lockbox.
+        Default: "airflow/variables"
+    :param config_prefix: Specifies the prefix of the secret to read to get Configurations.
+        If set to None (null in JSON), requests for variables will not be sent to Yandex Lockbox.
+        Default: "airflow/config"
+    :param sep: Specifies the separator used to concatenate secret_prefix and secret_id.
+        Default: "/"
+    :param endpoint: Specifies an API endpoint.
+        Leave blank to use default.
+    """
+
+    def __init__(
+        self,
+        yc_oauth_token: str | None = None,
+        yc_sa_key_json: str | None = None,
```

Comment:
What if make this parameter either python dictionary or string? In this case users might avoid escaping this parameter in `[secrets] backend_kwargs`

https://github.com/apache/airflow/blob/b1ce06ec54803be2768bb0b68db4e21dab119674/docs/apache-airflow-providers-yandex/secrets-backends/yandex-cloud-lockbox-secret-backend.rst?plain=1#L126-L131

Reply:
wow 😱